dhall-docs/src/Dhall/Docs/Core.hs

```diff
@@ -128,90 +168,69 @@ getAllDhallFiles baseDir = do
 
             Only the "global" level of the file is analyzed
         -}
-
         getLetBindingsWithName :: Text -> [Binding Void Import]
-        getLetBindingsWithName name = filter bindName $ reverse bindings
+        getLetBindingsWithName name = filter bindName $ reverse $ bindings expr
           where
-            bindings :: [Binding Void Import]
-            bindings = case expr of
-                Let b@Binding{} e ->
-                    let MultiLet bs _ = Dhall.Core.multiLet b e
-                    in NonEmpty.toList bs
-                _ -> []
             bindName (Binding _ x _ _ _ _) = x == name
 
 
         getLetBindingWithIndex :: Int -> [Binding Void Import] -> Maybe (Binding Void Import)
-        getLetBindingWithIndex i bindings =
-            case drop i bindings of
+        getLetBindingWithIndex i bs =
+            case drop i bs of
                 [] -> Nothing
                 binding : _ -> Just binding
 
-{-| Calculate the relative path needed to access files on the first argument
-    relative from the second argument.
+    examplesFromAssertions :: Expr Void Import -> [Expr Void Import]
+    examplesFromAssertions expr = Maybe.mapMaybe fromAssertion values
+      where
+        values :: [Expr Void Import]
+        values = map value $ bindings expr
 
-    The second argument needs to be a child of the first, otherwise it will
-    loop forever
+        fromAssertion :: Expr Void Import -> Maybe (Expr Void Import)
+        fromAssertion (Assert e) =  Just e
+        fromAssertion _ = Nothing
 
-    Examples:
+{-| Given a relative path, returns as much @..\/@ misdirections as needed
+    to go to @.@
 
->>> resolveRelativePath [absdir|/a/b/c/|] [absdir|/a/b/c/d/e|]
-"../../"
->>> resolveRelativePath [absdir|/a/|] [absdir|/a/|]
+>>> resolveRelativePath [reldir|.|]
 ""
+>>> resolveRelativePath [reldir|a|]
+"../"
+>>> resolveRelativePath [reldir|a/b/c|]
+"../../../"
 -}
-resolveRelativePath :: Path Abs Dir -> Path Abs Dir -> FilePath
-resolveRelativePath outDir currentDir =
-    if outDir == currentDir then ""
-    else "../" <> resolveRelativePath outDir (Path.parent currentDir)
+resolveRelativePath :: Path Rel Dir -> FilePath
+resolveRelativePath currentDir =
+    case FilePath.dropTrailingPathSeparator $ Path.fromRelDir currentDir of
+        "." -> ""
```

Comment:
What about a path that already has `..` as a path component, such as `a/..`?

Reply:
paths from the `path` package never contains those.